pkg/node/manager/manager.go

```diff
@@ -338,7 +342,7 @@ func (m *manager) isSelectedForManagement(v1node *v1.Node) bool {
 		return false
 	}
 
-	return isWindowsNode(v1node) || canAttachTrunk(v1node)
+	return m.conditions.IsWindowsIPAMEnabled() || canAttachTrunk(v1node)
```

Comment:
shouldn't this be `isWindowsNode(v1node) && m.conditions.IsWindowsIPAMEnabled()`?

Reply:
@sushrk you may want to address this as well. Thanks!

Reply:
Fixed.